File: modules/common/src/main/java/com/xiaoxin/common/utils/GsonUtils.java
package com.xiaoxin.common.utils;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.reflect.TypeToken;

import java.util.List;

/**
 * @author: Admin
 * @date: 2021-12-02
 */
public class GsonUtils {

    static Gson gson = new GsonBuilder().create();

    public static String toJson(Object obj) {
        return gson.toJson(obj);
    }

    public static <T> T toObject(String resJson) {
        T data = gson.fromJson(resJson,new TypeToken<T>(){}.getType());
        return data;
    }

    public static <T> List<T> toListObject(String resJson) {
        return gson.fromJson(resJson,new TypeToken<List<T>>(){}.getType());
    }

}

File: basic/src/main/java/com/xiaoxin/basic/utils/GsonUtils.kt
package com.xiaoxin.basic.utils

import com.google.gson.GsonBuilder
import com.google.gson.reflect.TypeToken
import java.lang.reflect.Type

/**
 * @author: Admin
 * @date: 2021-12-02
 */
object GsonUtils {

    var gson = GsonBuilder().create()
    fun toJson(obj: Any?): String {
        return gson.toJson(obj)
    }

    fun <T> toObject(resJson: String?): T {
        return gson.fromJson(resJson.toString(), object : TypeToken<T>() {}.type)
    }

    fun <T> toObject(resJson: String?, type: Type): T {
        return gson.fromJson(resJson, type)
    }

}